File: MobyShop.Model/Models/Artikli.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MobyShop.Models
{
    public class Artikli
    {
        public int ArtikalId { get; set; }
        public string Naziv { get; set; }
        public string Sifra { get; set; }
        public decimal Cijena { get; set; }
        public byte[] Slika { get; set; }
        public byte[] SlikaThumb { get; set; }
        public bool? Status { get; set; }
        public int KarakteristikeId { get; set; }
        public int ModelId { get; set; }
        public int ProizvodjacId { get; set; }

        public string Proizvodjac { get; set; }
        public string Model { get; set; }


        public decimal ProsjecnaOcjena { get; set; }
    }
}

File: MobyShop.Mobile/lib/models/Artikli.dart
import 'dart:convert';
import 'dart:typed_data';

class Artikli {
  final int? artikalId;
  final String? naziv;
  final String? cijena;
  final List<int>? slika;
  final String? proizvodjac;
  final String? model;
  final bool? status;
  final int? modelId;

  Artikli({
    this.artikalId,
    this.naziv,
    this.cijena,
    this.slika,
    this.proizvodjac,
    this.model,
    this.status,
    this.modelId
  });

  factory Artikli.fromJson(Map<String, dynamic> json){
    String stringByte = json["slika"] as String;
    List<int>? bytes=base64.decode(stringByte);
    return Artikli(
      artikalId:int.parse(json["artikalId"].toString()),
      naziv: json["naziv"],
      cijena: json["cijena"].toString(),
      slika: bytes,
      proizvodjac: json["proizvodjac"].toString(),
      model: json["model"].toString(),
      status: json["status"],
      modelId:  json["modelId"] as int,
    );
  }

  Map<String, dynamic> toJson() => {
    "artikalId": artikalId,
    "naziv": naziv,
    "cijena": cijena,
    "slika": slika,
    "proizvodjac":proizvodjac,
    "model":model,
    "status": status,
    "modelId":modelId
  };
}